src/wallet/scriptpubkeyman.cpp

```diff
@@ -2009,9 +2009,15 @@ std::optional<MigrationData> LegacyScriptPubKeyMan::MigrateToDescriptor()
 }
 
 bool LegacyScriptPubKeyMan::DeleteRecords()
+{
+    return RunWithinTxn(m_storage.GetDatabase(), /*process_desc=*/"delete legacy records", [&](WalletBatch& batch){
+        return DeleteRecords(batch);
+    });
+}
```

Comment:
In cd694033173ec7de2a16187432a3d7aa33ff62ca "wallet: introduce 'DeleteRecords(batch)' to be used within an external db txn"

After this PR, this `DeleteRecords()` overload is only used by the tests, so I think it can just be removed (and perhaps never introduced).